src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -241,19 +249,24 @@ def reconstruct_messages(
                             exceeded_org_quotas += 1
                         continue
 
-                    if new_v is None:
-                        metadata = bulk_record_meta[org_id].get(v)
-                        if (
-                            metadata
-                            and metadata.fetch_type_ext
-                            and metadata.fetch_type_ext.is_global
-                        ):
-                            exceeded_global_quotas += 1
+                    value_to_write = v
```

Comment:
I think we need a refactoring of this method. 
See, this seems suspicious:
- this method is 150 lines long.
- The same condition `if self.__should_index_tag_values` is there twice far away from each other.
This makes it very hard to grasp logically what is the difference between the two modes and very easy to break (updating one without the other).

I think a good followup task would be to separate the logic that branches between the two modes from the logic that actually does stuff, from the logic that manages exceptions so that the two code paths are clear.

- Error management. This is easy. python has Exceptions. Let's have a method that reconstruct the message and raise exceptions when something is wrong, and another one that wraps it and decides what to do with them. This removes the logging logic and the logic to remove a message from this method.
- we can have a version 1 method and a version 2 method. One deals with indexed tags the other does not.
- These methods would have a lot of duplication, which means there is a missing abstraction there to manipulate each individual message. This one would require some additional thinking.